client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -107,13 +107,26 @@ public boolean deleteStream(String scopeName, String streamName) {
         return  Futures.getThrowingException(controller.deleteStream(scopeName, streamName));
     }
 
+    @Override
+    public Iterator<String> listScopes() {
+        log.info("Listing scopes");
+        AsyncIterator<String> asyncIterator = controller.listScopes();
+        return new BlockingAsyncIterator<>(asyncIterator);
```

Comment:
`asyncIterator.asIterator()` will do the same, but you won't need to explicitly use `BlockingAsyncIterator`. This will help in case we want to change the implementation later.